### ISSUE TITLE
Clarification about register buffer weights

### ISSUE THREAD

**Issue by FrancescoMandru** [lyft/l5kit#305]
Hello, in the planning model baseline you do the following for the target availabilities weights:

```python
self.register_buffer("weights_scaling", torch.tensor(weights_scaling))
```

I wonder why you do that and what happens if you don't do that. For example I compute the loss function externally and for weights I do the following: 

```python
target_weights = (target_avail * torch.tensor([1.0, 1.0]).cuda()).view(target_traj.size(0), -1) # [batch_size, future_len*2]
```

From what I have understood, if you have parameters in your model, which should be saved and restored in the state_dict, but not trained by the optimizer, you should register them as buffers. This means that the two provided examples are the same

**Comment by lucabergamini** [comment]
Almost, the difference is in your `.cuda()` call. If you set the tensor with `register_buffer` it will be moved between devices with the model, if not it will be on you to ensure it is on the right device.

In general, if you use a tensor only inside the model methods it makes sense to have it as a buffer
